### PR TITLE
Math Magicians: tests

### PR DESCRIPTION
**Pull request by Kayonga99**
## The following were Implemented:

- No linter errors.
- Use correct git-flow.
- Set up React Testing Library.
- Add unit tests for the files operate.js and calculate.js using Jest.
- Create unit tests for all React components:
   - Use Jest snapshots to test the components.
  - Use React Testing Library.

### CODE REVIEW

File: src/App.test.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import { BrowserRouter as Router } from 'react-router-dom';
+import renderer from 'react-test-renderer';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom/extend-expect';
+import Calculator from './components/Calculator';
+import Home from './components/home';
+import NavBar from './components/navbar';
```
Comment (ShahierNashaat) [inline src/App.test.js]: - [ ] Kindly write the path of the `NavBar` file right like that.
```javascript
import NavBar from './components/navBar';
```

File: src/App.test.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    expect(tree).toMatchSnapshot();
+  });
+  test('Simulation of user clicking button 9 twice', () => {
+    render(<Calculator />);
+    const buttonEl = screen.getByText(/9/i);
+    userEvent.click(buttonEl);
+    userEvent.click(buttonEl);
+    expect(screen.getByText(/99/i)).toBeInTheDocument();
+  });
+});
```
Comment (ShahierNashaat) [inline src/App.test.js:121]: - [ ] [Optional] Kindly be noticed that it is better to write the tests for each component or file in a separate file.

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @Kayonga99, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- All tests are passing ✔️
- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Good overall design ✔️
- Your code is clean and well documented ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.test.js (hunk 1 above)
Comment (Kayonga99) [reply src/App.test.js]: Hey @ShahierNashaat I think the path is correctly imported I tried importing it this way `import NavBar from './components/navBar';` and got some errors.

**Review by descholar-ceo — APPROVED** [review]
Hi @Kayonga99 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented required changes ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**